scripts/build-intl-polyfills.mjs

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env node
+
+import fs from 'node:fs';
+
+import {supportedLocales as relativeTimeFormatSupportedLocales} from '@formatjs/intl-relativetimeformat/supported-locales.generated.js';
+import {supportedLocales as pluralRulesSupportedLocales} from '@formatjs/intl-pluralrules/supported-locales.generated.js';
+
+import languages from '../src/frontend/languages.json' assert {type: 'json'};
+import messages from '../translations/messages.json' assert {type: 'json'};
+
+build();
+
+function build() {
+  const locales = getPolyfillableLocales();
+
+  const outputPath = new URL(
+    '../src/frontend/polyfills/intl.ts',
+    import.meta.url,
+  ).pathname;
+
+  writePolyfillFile(locales, outputPath);
+
+  console.log(`Successfully added intl polyfill imports to ${outputPath}`);
+}
+
+/**
+ * Calculate set of locales that can be polyfilled by the relevant @formatjs modules
+ * @returns {Array<string>}
+ */
+function getPolyfillableLocales() {
+  const comapeoSupportedLocales = Object.keys(messages).filter(locale => {
+    const hasAtLeastOneTranslatedString =
+      Object.keys(messages[locale]).length > 0;
+
+    const hasTranslatedLanguageName = !!languages[locale];
+
+    return hasAtLeastOneTranslatedString && hasTranslatedLanguageName;
+  });
+
+  const localesToPolyfill = [];
+
+  for (const locale of comapeoSupportedLocales) {
+    const canPolyfillLocale = isFullySupportedLocale(locale);
+
+    if (!canPolyfillLocale) {
+      console.warn(`Cannot polyfill data for locale: ${locale}`);
+      continue;
+    }
+
+    localesToPolyfill.push(locale);
+  }
+
+  return localesToPolyfill;
+}
+
+/**
+ * Writes file containing polyfill imports
+ * @param {Array<string>} locales
+ * @param {string} outputPath
+ */
+function writePolyfillFile(locales, outputPath) {
+  const writer = fs.createWriteStream(outputPath, {flags: 'w'});
+
+  writer.write(
+    '// This file is automatically generated through scripts/build-intl-polyfills.mjs. Do not edit this directly!\n// If you need to rebuild it, you can run `npm run build:intl-polyfills`\n',
+  );
+
+  // Write lines to load base polyfills
+  writer.write(
+    createImportStatement('@formatjs/intl-getcanonicallocales/polyfill'),
+  );
+  writer.write(createImportStatement('@formatjs/intl-locale/polyfill'));
+
+  writer.write('\n');
+
+  // Write lines to load plural rules polyfill
+  writer.write(createImportStatement('@formatjs/intl-pluralrules/polyfill'));
+  for (const locale of locales) {
+    writer.write(
+      createImportStatement(`@formatjs/intl-pluralrules/locale-data/${locale}`),
+    );
+  }
```

Comment:
I am not sure if this matters, but perhaps we should check if the [locale needs to be polyfilled](https://formatjs.io/docs/polyfills/intl-pluralrules#dynamic-import--capability-detection), and only write the import statement as necessary.

Its working now, but would creating a polyfill if it is not needed cause a problem?

Reply:
the `shouldPolyfill()` functions provided for each of these won't make sense to use in this context because it would be checking for properties in the development node runtime, and not the runtime that exists on the device. 

https://github.com/formatjs/formatjs/blob/main/packages/intl-pluralrules/should-polyfill.ts

> Its working now, but would creating a polyfill if it is not needed cause a problem?

a properly implemented polyfill is relatively inexpensive and should not cause any side effects if it's not needed. they do a proper job of this based on the implementations i'm seeing (for example, for plural rules: https://github.com/formatjs/formatjs/blob/main/packages/intl-pluralrules/polyfill.ts). writing the import lines into the file even if the runtime eventually gets built-in support is not problematic, as the polyfill will then no-op.

the polyfills check for two things:

1. does the interface in question already exist or not in the runtime?
2. does formatjs provide support for the locale(s) specified

In our case, only (2) really matters when we're running this as a build script. as a result, it makes more sense to only check the supported locales here.